src/api/java/com/minecolonies/api/entity/SpearEntity.java

```diff
@@ -0,0 +1,154 @@
+package com.minecolonies.api.entity;
+
+import com.minecolonies.api.items.ModItems;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.AbstractArrowEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.IPacket;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.IndirectEntityDamageSource;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.EntityRayTraceResult;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import net.minecraftforge.fml.network.NetworkHooks;
+import org.jetbrains.annotations.NotNull;
+
+public class SpearEntity extends AbstractArrowEntity implements ICustomAttackSound
```

Comment:
missing javadoc on class and vars

Reply:
The projectile should get a limited lifetime similar to our existing CustomArrowEntity, to make sure it'll despawn, even if it never hits the ground